lib/connection.ts

```diff
@@ -166,6 +167,20 @@ export class Connection {
     );
   }
 
+  suspendReconnectUntil(suspendPromise: Promise<void>) {
+    this.suspendReconnectPromise = suspendPromise;
+  }
+
+  suspend(suspendPromise?: Promise<void>) {
```

Comment:
Can we have a default FALSE promise?
When we want just suspend the reconnect without any special logic, just call the suspend() and this will suspend the reconnect process.

Reply:
How would the code then know when to unsuspend?

Reply:
Yea, good question, I just think about my scenario with want create a new connection. You right its not good idea.

Reply:
Or maybe we can add another function to unsuspend? I do not know its good idea or not, But this will like as pause and resume. suspend promise is like as an advanced logic to suspend the reconnect process.

Reply:
You can easily do that yourself, check 

https://github.com/home-assistant/frontend/pull/6149